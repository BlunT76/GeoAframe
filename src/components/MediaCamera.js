import React, { Component } from 'react';
import Webcam from 'react-webcam';

const videoContainer = {
  display: 'block',
  height: '100%',
  overflow: 'hidden',
  width: '100%',
  zIndex: '0',
};

class MediaCamera extends Component {
  render() {
    const videoConstraints = {
      facingMode: 'environment',
    };
    return (
      <Webcam
        style={videoContainer}
        audio={false}
        videoConstraints={videoConstraints}
        ref={node => (this.webcam = node)}
      />
    );
  }
}

export default MediaCamera;
