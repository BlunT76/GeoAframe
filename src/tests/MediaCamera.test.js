import React from 'react';
import renderer from 'react-test-renderer';
import MediaCamera from '../components/MediaCamera';

test('MediaCamera is always the same', () => {
  const component = renderer.create(
    <MediaCamera />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
