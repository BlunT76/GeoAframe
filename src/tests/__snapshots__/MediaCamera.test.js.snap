// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCamera is always the same 1`] = `
<video
  autoPlay={true}
  className="videoContainer"
  height={480}
  muted={false}
  playsInline={true}
  width={640}
/>
`;
